pkg/asset/cluster/tfvars.go

```diff
@@ -510,17 +510,38 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			caCert = string(caFile)
 		}
 
-		masters, err := mastersAsset.Machines()
-		if err != nil {
-			return err
+		var masterSpecs []*openstackprovider.OpenstackProviderSpec
+		{
+			masters, err := mastersAsset.Machines()
+			if err != nil {
+				return err
+			}
+
+			for _, master := range masters {
+				masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
```

Comment:
In general, are we ok with a crash in the event that ProviderSpec is unset, or of an unexpected type?

Reply:
I believe that at this point, that property should be exactly of type `*openstackprovider.OpenstackProviderSpec`.

The alternative would be a controlled type assertion, but then what do we do in case it's not the type we expect? The answer heavily depends on how the user got to that state, which is information I don't know how to get.